main.go

```diff
@@ -3,16 +3,41 @@ package main
 import (
 	"fmt"
 	"log"
+	"os"
+	"slices"
 )
 
 const gomodName = "go.mod"
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
 
+func run() error {
 	updatedContents, err := MergeRequires(gomodName)
 	if err != nil {
-		log.Fatal(err)
+		return fmt.Errorf("failed to merge requires: %w", err)
+	}
+
+	// check if we want to replace the contents of go.mod
+	if !slices.Contains(os.Args, "--replace") {
```

Comment:
I would prefer if you could use a flag instead of doing that.

https://pkg.go.dev/flag

Reply:
Can we use `--in-place` instead of `--replace`?

Using `flags` package is a good suggestion since:
 
- Adding future flags becomes easier
- Can add aliases easily. I was thinking something like `-i` similar to `sed`, `perl` and `yq`. The latter also uses `--in-place`.

Reply:
I also think the condition should be reversed and the file update should be inside this `if` block. 

I'm against an early return here since 
- the default behaviour of the binary is to print to output, which will be reflected in code,
- we don't have to worry about the `if` block getting crowded since it's just two operations (get the mode and update the file).

Reply:
Updated to use the `flag` package and `-i` and `--in-place`

also switched the if condition to make it more readable

thanks for the suggestions

Reply:
I don't think it warrants an external dependency just yet, so it's fine for now.